sov-modules/sov-app-template/src/lib.rs

```diff
@@ -155,27 +151,21 @@ where
                         batch_workspace = batch_workspace.commit();
                     }
                     Err(e) => {
-                        // TODO: all the previous "successful" txs will be reverted, is that ok?
                         self.revert_and_slash(batch_workspace);
                         panic!("Demo app txs must succeed but failed with err: {}", e)
                     }
                 }
             } else {
                 // If the serialization is invalid, the sequencer is malicious. Slash them.
-                // TODO all the previous "successful" txs will be reverted, is that ok?
                 self.revert_and_slash(batch_workspace);
-                return Err(ConsensusSetUpdate::slashing(sequencer));
+                anyhow::bail!("Tx decoding error");
             }
         }
 
         // TODO: calculate the amount based of gas and fees
-        if self
-            .tx_hooks
+        self.tx_hooks
             .exit_apply_batch(0, &mut batch_workspace)
-            .is_err()
-        {
-            // TODO handle error: https://github.com/Sovereign-Labs/sovereign/issues/174
-        }
+            .unwrap();
```

Comment:
Are we sure we want to `panic` on failure here? If so, let's add a comment explaining or use `expect`

Reply:
I added `?` so the error will be handled like all the other error.